Explorer/Assets/DCL/CharacterPreview/CharacterPreviewController.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Arch.Core;
+using DCL.AvatarRendering.AvatarShape.Components;
+using DCL.AvatarRendering.Wearables;
+using DCL.AvatarRendering.Wearables.Components;
+using DCL.AvatarRendering.Wearables.Components.Intentions;
+using DCL.AvatarRendering.Wearables.Helpers;
+using DCL.Optimization.Pools;
+using ECS.LifeCycle.Components;
+using ECS.Prioritization.Components;
+using ECS.StreamableLoading.Common;
+using ECS.Unity.Transforms.Components;
+using System;
+
+namespace DCL.CharacterPreview
+{
+    public readonly struct CharacterPreviewController : IDisposable
+    {
+        private readonly World globalWorld;
+        private readonly Entity characterPreviewEntity;
+        private readonly CharacterPreviewContainer characterPreviewContainer;
+        private readonly CharacterPreviewCameraController cameraController;
+        private readonly IComponentPool<CharacterPreviewContainer> characterPreviewContainerPool;
+
+        public CharacterPreviewController(World world, CharacterPreviewContainer container, CharacterPreviewInputEventBus inputEventBus, IComponentPool<CharacterPreviewContainer> characterPreviewContainerPool)
+        {
+            globalWorld = world;
+            characterPreviewContainer = container;
+            cameraController = new CharacterPreviewCameraController(inputEventBus, characterPreviewContainer);
+            this.characterPreviewContainerPool = characterPreviewContainerPool;
+
+            // TODO add meaningful ID and Name
+
+            characterPreviewEntity = world.Create(
+                new TransformComponent(container.avatarParent),
+                new AvatarShapeComponent("CharacterPreview", "CharacterPreview"));
+        }
+
+        public void UpdateAvatar(CharacterPreviewModel model)
+        {
+            if (globalWorld == null || model.Wearables == null || model.Wearables.Count <= 0) return;
+
+            ref AvatarShapeComponent avatarShape = ref globalWorld.Get<AvatarShapeComponent>(characterPreviewEntity);
+
+            avatarShape.SkinColor = model.SkinColor;
+            avatarShape.HairColor = model.HairColor;
+            avatarShape.BodyShape = BodyShape.FromStringSafe(model.BodyShape);
+
+            if (!avatarShape.WearablePromise.IsConsumed)
```

Comment:
You don't have to check it - there is right the same check in the implementation of `ForgetLoading`